src/lib.rs

```diff
@@ -126,12 +126,74 @@ pub struct SelfEncryptor<'a> {
 }
 
 impl<'a> SelfEncryptor<'a> {
-  /// This is the only constructor, for encryptor object
+  /// This is the only constructor, for encryptor object.
   /// Each SelfEncryptor is used for a single file.
   /// The parameters are a DataMap and Storage.
-  /// If new file use DataMap::None as first parameter
+  /// If new file, use DataMap::None as first parameter.
   /// The get and put of Storage need to be implemented to
   /// allow the SelfEncryptor to store encrypted chunks and retrieve when necessary.
+  // # Example usage
+  // ```rust
+  // let mut se = SelfEncryptor::new(&mut my_storage as MyStorage, datamap::DataMap::None);
+  // ```
+  //
+
+/// # Example
+///
+/// SelfEncryptor Example
+///
+/// ```rust
+/// extern crate self_encryption
+/// extern crate rand;
+/// use use self_encryption::*;
+///
+/// fn main() {
```

Comment:
Hi David,
Hope weekend went well. Productive weekend my end. So I've implemented the
dotcopts interface for user input. I haven't tried compiling yet as I'd
need the office for that but I wondered if I should see with you to just
have a look on logic or post on slack. I notice Rhyman is great at
detecting whether something will work entirely by looking at code. But the
implementation seems good.
Anyways I also made good progress on random write test (who knew
implementing a subtract method in rust was a nightmare), but I am a bit
unsure on how to implement the vector pair in the C++ example. I will ask
on slack if I don't get it sorted tonight. I will have a look at C++
example for benchmark but probably compression should be next on my agenda.
But anyways yea good weekend work so happy.
Best wishes
On 20 Mar 2015 18:40, "David Irvine" <notifications@github.com> wrote:

> In src/lib.rs
> <https://github.com/dirvine/self_encryption/pull/17#discussion_r26865228>:
>
> > +  // # Example usage
> > +  // ```rust
> > +  // let mut se = SelfEncryptor::new(&mut my_storage as MyStorage, datamap::DataMap::None);
> > +  // ```
> > +  //
> > +
> > +/// # Example
> > +///
> > +/// SelfEncryptor Example
> > +///
> > +/// ```rust
> > +/// extern crate self_encryption
> > +/// extern crate rand;
> > +/// use use self_encryption::*;
> > +///
> > +/// fn main() {
>
> Again should this not take some user input to encrypt a file on disk or
> similar and then also to decrypt it. (see docopts lib for a nice
> interface).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dirvine/self_encryption/pull/17/files#r26865228>.
>